core/src/main/clojure/xtdb/trie.clj

```diff
@@ -176,3 +176,29 @@
                                              {:ordinal ordinal, :leaf node})))))}])))]
 
     (vec (trie-merge-tasks* (map #(some-> ^HashTrie % (.rootNode)) tries) []))))
+
+(defn- bucket-for [^bytes iid level]
+  (let [level-offset-bits (* HashTrie/LEVEL_BITS (inc level))
+        level-offset-bytes (/ (- level-offset-bits HashTrie/LEVEL_BITS) Byte/SIZE)]
+    (bit-and (bit-shift-right (get iid level-offset-bytes) (mod level-offset-bits Byte/SIZE)) HashTrie/LEVEL_MASK)))
+
+(defn iid-trie-merge-task [tries ^bytes iid]
```

Comment:
hoping that maybe when we reintroduce metadata this'll naturally Just Work™ as a metadata selector...?

(i.e. this is fine for now, just that it might be temporary)